native/cocos/core/assets/EffectAsset.h

```diff
@@ -36,15 +37,52 @@
 #include "renderer/core/PassUtils.h"
 #include "renderer/gfx-base/GFXDef.h"
 #include "renderer/pipeline/Define.h"
-
 namespace cc {
 
+#if !SWIGCOCOS
+
+template <typename T>
+struct IsVectorContainer : std::false_type {};
+
+template <typename E, typename A>
```

Comment:
It is better to have a new class to wrap these codes.

Reply:
Fixed